### DISCUSSION TITLE
Youtube Video on Homepage

### DISCUSSION THREAD

**Discussion by SingularDave** [thuliteio/doks#786] [Q&A] [answered]
Hello all. I found the shortcode for embedding a YouTube video (e.g. {{< youtube w7Ft2ymGmfc >}}). I'm trying to figure out how I can place a YouTube video on the Homepage in the "main" section. I've looked through the docs, and I can't figure it out. When I include the shortcode on the homepage, I get a "This content is blocked" response. Thanks for the help!

**Comment by jamie-moxon** [comment, marked as answer]
I tried it with the markdown code you wrote and it didn't work for me, because the home page is HTML.  I created a new div and got the YouTube embed code directly from the video and it worked

**Reply by SingularDave** [reply]
Thanks for the response. When I add it in with HTML, it functions well locally but doesn't when I deploy the site on CF Pages. It gives me the same "This content is blocked" response.

**Reply by h-enk** [reply]
Note that, Hugo processes shortcodes in Markdown (not HTML) — see also the Hugo docs: [What a Shortcode is](https://gohugo.io/content-management/shortcodes/#what-a-shortcode-is). So, add the YouTube shortcode to `./content/en/_index.md` (not `./layouts/index.html`).

For further instructions (and an example), see [How to add a YouTube video to Doks](https://github.com/h-enk/doks-786/blob/main/README.md#how-to-add-a-youtube-video-to-doks)

**Reply by SingularDave** [reply]
I figured out my issue. It was CSP issue. I fixed it. Works great now.
